Assets/Plugins/Editor/Uplift/UpliftManager.cs

```diff
@@ -79,13 +79,78 @@ public enum InstallStrategy {
             UPDATE_LOCKFILE
         }
 
+        public struct DependencyState
+        {
+            public DependencyDefinition definition;
+            public InstalledPackage installed;
+            public PackageRepo bestMatch, latest;
+            public bool transitive;
+        }
+        
         public void InstallDependencies(InstallStrategy strategy = InstallStrategy.UPDATE_LOCKFILE, bool refresh = false)
         {
             if (refresh) UpliftManager.ResetInstances();
             PackageRepo[] targets = GetTargets(GetDependencySolver(), strategy);
             InstallPackages(targets);
         }
 
+        public DependencyState[] GetDependenciesState(bool refresh = false)
+        {
+            if (refresh) UpliftManager.ResetInstances();
+            Upbring upbring = Upbring.Instance();
+            PackageRepo[] targets = GetTargets(GetDependencySolver(), InstallStrategy.UPDATE_LOCKFILE);
+
+            bool any_installed =
+                        upbring.InstalledPackage != null &&
+                        upbring.InstalledPackage.Length != 0;
+            
+            List<DependencyState> dependenciesState = new List<DependencyState>();
+            foreach(DependencyDefinition definition in upfile.Dependencies)
+                AppendDependencyState(
+                        ref dependenciesState,
+                        definition,
+                        targets,
+                        any_installed
+                    );
+
+            return dependenciesState.ToArray();
+        }
+
+        public void AppendDependencyState(
+            ref List<DependencyState> dependenciesState,
+            DependencyDefinition definition,
+            PackageRepo[] targets,
+            bool any_installed,
+            bool transitive = false
+        )
+        {
+            Upbring upbring = Upbring.Instance();
+
+            DependencyState state = new DependencyState
+            {
+                definition = definition,
+                latest = PackageList.Instance().GetLatestPackage(definition.Name)
+            };
+
+            state.bestMatch = targets.First(pr => pr.Package.PackageName == definition.Name);
```

Comment:
Given that we just refreshed, could that fail? I.e. `targets.First` return `null`?

Reply:
I'm not sure. We go through the dependencies as does the targets, so there should not be an issue. We can still check for null though, but what behaviour should we have then?

Reply:
We keep it as is